src/reducers/searchReducer.js

```diff
@@ -9,6 +9,8 @@ const initialState = {
 
 export default function Search(state = initialState, action) {
   switch (action.type) {
+    case 'SET_TOOL':
+      return (action.tool == 'search') ? state : initialState;
```

Comment:
ja som sa stretol, preto to budem opravovat (namiesto venovaniu sa novym fraturam ;-)).